src/main/java/io/mosip/mimoto/controller/IssuersController.java

```diff
@@ -78,4 +92,73 @@ public ResponseEntity<Object> getIssuerConfig(@PathVariable("issuer-id") String
+                                                         @PathVariable("credentials-supported-id") String credentialsSupportedId) {
+
+        ResponseWrapper<Object> responseWrapper = new ResponseWrapper<>();
+        responseWrapper.setId(ID);
+        responseWrapper.setVersion("v1");
+        responseWrapper.setResponsetime(DateUtils.getRequestTimeString());
+
+        try{
+            IssuerDTO issuerConfig = issuersService.getIssuerConfig(issuerId);
+            CredentialIssuerWellKnownResponse credentialIssuerWellKnownResponse = issuersService.getCredentialWellKnownFromJson();
+            Optional<CredentialsSupportedResponse> credentialsSupportedResponse = credentialIssuerWellKnownResponse.getCredentials_supported().stream()
+                    .filter(credentialsSupported -> credentialsSupported.getId().equals(credentialsSupportedId))
+                    .findFirst();
+            if (credentialsSupportedResponse.isEmpty()){
+                logger.error("Invalid credential supported id passed - {}", credentialsSupportedId);
+                responseWrapper.setErrors(List.of(new ErrorDTO(INVALID_CREDENTIAL_SUPPORTED_TYPE_ID_EXCEPTION.getCode(), INVALID_CREDENTIAL_SUPPORTED_TYPE_ID_EXCEPTION.getMessage())));
+                return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(responseWrapper);
+            }
+            ByteArrayInputStream inputStream =  issuersService.generatePdfForVerifiableCredentials(token, issuerConfig, credentialsSupportedResponse.get(), credentialIssuerWellKnownResponse.getCredential_endpoint());
+            //PDF file name with issuer display name and credential supported type display name
+            String pdfFileName = issuerConfig.getDisplay().stream().filter(displayDTO -> displayDTO.getLanguage().equals("en")).map(DisplayDTO::getName).findFirst().orElse(null) +
+                    "_" + credentialsSupportedResponse.get().getDisplay().stream()
+                    .filter(credentialSupportedDisplayResponse -> credentialSupportedDisplayResponse.getLocale().equals("en")).map(CredentialSupportedDisplayResponse::getName).findFirst().orElse(null);
+            return ResponseEntity
+                    .ok()
+                    .contentType(MediaType.APPLICATION_PDF)
+                    .header(HttpHeaders.CONTENT_DISPOSITION, String.format("attachment; filename=%s.pdf", pdfFileName))
+                    .header(HttpHeaders.ACCESS_CONTROL_EXPOSE_HEADERS, "Content-Disposition")
+                    .body(new InputStreamResource(inputStream));
+        }catch (ApiNotAccessibleException | IOException exception){
+            logger.error("Exception occurred while fetching issuer metadata ", exception);
+            responseWrapper.setErrors(List.of(new ErrorDTO(API_NOT_ACCESSIBLE_EXCEPTION.getCode(), API_NOT_ACCESSIBLE_EXCEPTION.getMessage())));
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(responseWrapper);
```

Comment:
can we change the HttpStatus here to BAD_REQUEST ?

Reply:
changed this